src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheck.java

```diff
@@ -92,32 +102,87 @@
  * </p>
  * <pre>
  * for(int i = 1; i &lt; 10; i++) {}
+ * for(int K = 1; K &lt; 10; K++) {}
+ * List list = new ArrayList();
+ * for (Object o : list) {}
+ * for (Object O : list) {}
  * </pre>
  * <p>
  * An example of how to configure the check to allow one character variable name in
  * <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html">
- * initialization expressions</a> in FOR loop:
+ * initialization expressions</a> in FOR loop, where regexp allows 2 or more chars:
  * </p>
  * <pre>
  * &lt;module name="LocalVariableName"&gt;
+ *   &lt;property name="format" value="^[a-z][_a-zA-Z0-9]+$"/&gt;
  *   &lt;property name="allowOneCharVarInForLoop" value="true"/&gt;
  * &lt;/module&gt;
  * </pre>
- *
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   void MyMethod() {
+ *     int good = 1;
+ *     int g = 0; // violation
+ *     for (int v = 1; v &lt; 10; v++) { // OK
+ *         int a = 1; // violation
+ *     }
+ *     for (int V = 1; V &lt; 10; V++) { // OK
+ *         int I = 1; // violation
+ *     }
+ *     List list = new ArrayList();
+ *     for (Object o : list) { // OK
+ *         String a = ""; // violation
+ *     }
+ *     for (Object O : list) { // OK
+ *         String A = ""; // violation
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * An example of how to configure the check to that all variables have 3 or more chars in name:
+ * </p>
+ * <pre>
+ * &lt;module name="LocalVariableName"&gt;
+ *   &lt;property name="format" value="^[a-z][_a-zA-Z0-9]{2,}$"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   void MyMethod() {
+ *     int goodName = 0;
+ *     int i = 1; // violation
+ *     for (int var = 1; var &lt; 10; var++) { //OK
+ *       int j = 1; // violation
+ *     }
+ *   }
+ * }
+ * </pre>
  * @since 3.0
  */
 public class LocalVariableNameCheck
     extends AbstractNameCheck {
 
-    /** Regexp for one-char loop variables. */
-    private static final Pattern SINGLE_CHAR = Pattern.compile("^[a-z]$");
-
     /**
      * Allow one character variable name in
      * <a href="https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html">initialization expressions</a>
-     * in FOR loop. For example:
+     * in FOR loop if one char variable name is prohibited by {@code format} regexp. For example:
      * <pre>
-     * for (int i = 1; i &lt; 10; i++) {}
+     * for (int i = 1; i &lt; 10; i++) { // OK
+     *     int j = 1; // violation
+     * }
+     * for (int K = 1; K &lt; 10; K++) { // OK
+     *     int j = 1; // violation
+     * }
+     * List list = new ArrayList();
+     * for (Object o : list) { // OK
+     *     int j = 1; // violation
+     * }
+     * for (Object O : list) { // OK
+     *     int j = 1; // violation
+     * }
```

Comment:
@strkkk , is it something(CI) force you to update doc on field ?
just curious ...

I do not think we should have very extensive jaavadocs over properties, we have dedicated area in Class javadoc for this, no need to copy examples in other parts.

@pbludov , fyi.

Reply:
I agree with @romani, this long example shouldn't be in property description. That is why it is being forced on field through our xdoc / javadoc alignment process.

Reply:
@romani yes, there is fail if javadoc is updated only for class, not for field/setter

Reply:
if any code changes be required, I would be good to remove code example from property description. Examples section should be enough, and we have link to java documentation , it is enough.